File: activity/app/src/main/java/com/rikkei/training/activity/activity/AActivity.java
package com.rikkei.training.activity.activity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.rikkei.training.activity.MediaManager;
import com.rikkei.training.activity.R;
import com.rikkei.training.activity.base.BaseActivity;

public class AActivity extends BaseActivity implements View.OnClickListener {
    private final String TAG = AActivity.class.getSimpleName();
    private Button mButtonMain;
    private Button mButtonAB;
    private Button mButtonBForResult;
    private MediaManager media;
    private String link = "/storage/emulated/0/Download/Ket Thuc Lau Roi - Le Bao Binh.mp3";
    public final int REQUEST_A = 5;

    @Override
    protected void initViews() {
        Log.d(TAG, "on Create");
        mButtonMain = findViewById(R.id.btn_act_main);
        mButtonAB = findViewById(R.id.btn_act_ab);
        mButtonBForResult = findViewById(R.id.btn_str_b_for_result);
        mButtonAB.setOnClickListener(this);
        mButtonMain.setOnClickListener(this);
        mButtonBForResult.setOnClickListener(this);
    }

    @Override
    protected void initData() {
        media = new MediaManager(link, this);
        media.create();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_a;
    }

    @Override
    protected void onStart() {
        super.onStart();
        media.start();
        Log.d(TAG, "on Start");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "on Resume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        media.pause();
        Log.d(TAG, "on Pause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "on Stop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "on Destroy");
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btn_act_main:
                media.stop();
                finish();
                break;
            case R.id.btn_act_ab:
                Intent intent = new Intent(this, BActivity.class);
                startActivity(intent);
                break;
            case R.id.btn_str_b_for_result:
                Intent intentForResult = new Intent(this, BActivity.class);
                startActivityForResult(intentForResult, REQUEST_A);
                break;
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == REQUEST_A) {
            if (resultCode == RESULT_OK) {
                Toast.makeText(this, "request success", Toast.LENGTH_SHORT).show();
            } else {
                Toast.makeText(this, "result false", Toast.LENGTH_SHORT).show();
            }
        } else {
            Toast.makeText(this, "request false", Toast.LENGTH_SHORT).show();
        }
    }
}

File: Activitykotlin/app/src/main/java/com/rikkei/training/activitykotlin/AActivity.kt
package com.rikkei.training.activitykotlin

import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.activity_a.*

class AActivity : AppCompatActivity(), View.OnClickListener {
    private var link = "/storage/emulated/0/Download/Di Du Dua Di - Bich Phuong.mp3"
    private var media: MediaManager? = null
    private var data: SystemData? = null
    private val mPERMISSION: List<String> =
        listOf(android.Manifest.permission.READ_EXTERNAL_STORAGE)

    override fun onClick(p0: View?) {
        when (p0!!.id) {
            R.id.btn_act_b -> toActA()
            R.id.btn_act_rotate -> toActR()
        }
    }

    private fun toActA() {
        media!!.pause()
        val intent = Intent(this, BActivity::class.java)
        startActivity(intent)
    }

    private fun toActR() {
        media!!.stop()
        val intent = Intent(this, RotateActivity::class.java)
        startActivity(intent)
    }

    private val mTAG = "AActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(mTAG, "on Create")
        if (checkPermission()) {
            initViews()
        }
    }

    private fun checkPermission(): Boolean {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            for (p in mPERMISSION) {
                if (checkSelfPermission(p) != PackageManager.PERMISSION_GRANTED) {
                    requestPermissions(mPERMISSION.toTypedArray(), 0)
                    return false
                }
            }
        }
        return true
    }

    private fun initViews() {
        data = SystemData(this)
        data!!.readData()
        media = MediaManager(this, link)
        setContentView(R.layout.activity_a)
        btn_act_a.setOnClickListener(this)
        btn_act_rotate.setOnClickListener(this)
        media!!.create()
    }

    override fun onStart() {
        media!!.start()
        super.onStart()
        Log.d(mTAG, "on Start")
    }

    override fun onResume() {
        super.onResume()
        Log.d(mTAG, "on Resume")
    }

    override fun onPause() {
        super.onPause()
        Log.d(mTAG, "on Pause")
    }

    override fun onStop() {
        super.onStop()
        Log.d(mTAG, "on Stop")
    }

    override fun onDestroy() {
        super.onDestroy()
        media!!.stop()
        Log.d(mTAG, "on Destroy")
    }
}
